pkg/handlers/v1/eni_test.go

```diff
@@ -0,0 +1,186 @@
+	for _, tt := range tc {
+		tt := tt
+		t.Run(tt.Name, func(t *testing.T) {
+			data, err := ioutil.ReadFile(filepath.Join("testdata", tt.InputFile))
+			require.Nil(t, err)
+
+			var input Input
+			err = json.Unmarshal(data, &input)
+			require.Nil(t, err)
+
+			transformer := &Transformer{LogFn: logFn}
+			output, err := transformer.Handle(context.Background(), input)
+			if tt.ExpectError {
+				require.NotNil(t, err)
+			} else {
+				require.Nil(t, err)
+			}
+
+			assert.Equal(t, tt.ExpectedOutput.AccountID, output.AccountID)
+			assert.Equal(t, tt.ExpectedOutput.Region, output.Region)
+			assert.Equal(t, tt.ExpectedOutput.ARN, output.ARN)
+			assert.Equal(t, tt.ExpectedOutput.ResourceType, output.ResourceType)
+			assert.Equal(t, tt.ExpectedOutput.ChangeTime, output.ChangeTime)
+			assert.True(t, reflect.DeepEqual(tt.ExpectedOutput.Changes, output.Changes), "The expected changes were different than the result")
+		})
+	}
+}
+
+func TestFilterENI(t *testing.T) {
+	filteredConfig := eniConfiguration{
+		Description:        "ELB app/never-used",
+		PrivateIPAddresses: nil,
+		RequesterID:        elbManaged,
+		RequesterManaged:   false,
+	}
+	jsonFilteredConfig, err := json.Marshal(filteredConfig)
+	if err != nil {
+		print("Could not marshal test struct")
+	}
+
+	filteredConfigItem := configurationItem{
+		AWSAccountID:                 "123456789",
+		ResourceType:                 "AWS::EC2::NetworkInterface",
+		ARN:                          "arn:aws:ec2:us-east-1:12345678910:network-interface/eni-hhhhhhh888888",
+		AWSRegion:                    "us-east-1",
+		ConfigurationItemCaptureTime: "2020-08-21T13:00:01.000Z",
+	}
+
+	filteredConfigEvent := awsConfigEvent{
+		ConfigurationItem: filteredConfigItem,
+	}
+
+	transformer := eniTransformer{}
+
+	t.Run("eni-created", func(t *testing.T) {
+		filteredConfigEvent.ConfigurationItem.Configuration = json.RawMessage(jsonFilteredConfig)
+
+		createOutput, err := transformer.Create(filteredConfigEvent)
+		assert.Nil(t, err)
+		assert.True(t, createOutput.Changes == nil, "Expected empty changes due to filtering")
+	})
+
+	t.Run("eni-deleted", func(t *testing.T) {
+		// Because deleting looks at the PreviousValue for filtering, we need some more set up
+		filteredConfigDiff := eniConfigurationDiff{PreviousValue: &filteredConfig}
+		jsonFilteredConfigDiff, err := json.Marshal(filteredConfigDiff)
+		if err != nil {
+			print("Could not marshal test struct")
+		}
+
+		filteredConfigEvent.ConfigurationItemDiff = configurationItemDiff{
+			ChangedProperties: map[string]json.RawMessage{
+				"Configuration": json.RawMessage(jsonFilteredConfigDiff),
+			},
+		}
+
+		deleteOutput, err := transformer.Delete(filteredConfigEvent)
+		assert.Nil(t, err)
+		assert.True(t, deleteOutput.Changes == nil, "Expected empty changes due to filtering")
+	})
+}
+
+func TestErrorENI(t *testing.T) {
+	malformedConfigItem := configurationItem{
+		// excluding AWSAccountID so that we can trip our checks in the transformer
```

Comment:
question: I am slightly unsure what `trip` means here, could you briefly explain?

Reply:
Yes, in all of our transformers we call getBaseOutput and check to see if any errors were returned. Errors would come from any of the awsConfigEvent fields that we consider essential (like AWSAccountID or region) being missing.
These tests are to get test coverage for the case where getBaseOutput would return an error, which the transformer then bubbles up.

Is there a word other than trip that would be clearer? I could rephrase it as "excluding AWSAccountId so that we can cause a missingField error to bubble up from the transformer"

Reply:
I understand the context now, I think it is fine as is